### ISSUE TITLE
Rename js_ specials to ir_ specials

### ISSUE THREAD

**Issue by jburse** [jburse/dogelog-moon#101]
Since we started supporting the Python platform the js_ prefix
for JavaScript is not anymore appropriate. It turns out that
the JavaScript and the Python platform are very

similar, so it would be more approriate to rename these to the 
`ir_` prefix. Where `ir_` stands for intermediate representation (IR).
These are built-ins that would deal with Albuferia instruction

streams, and since they are reimplemented as specials each
platform has its own set of built-ins, like JavaScript resides in 
special.js and Python resides in special.py.

Nevertheless the Prolog code can refer to these through
the ir_ prefix, practically being agnostic to the platform. The 
change should also affect the js_clause, etc compounds.

**Comment by jburse** [comment]
There is also a set of dynamic database built-ins to 
help build the ISO core standard dynamic database predicates.
It possibly makes sense to change their prefix from js_ specials

to kb_ specials. Since they dont deal with the intermediate
representation (IR) of clauses itself, but rather with the data structures
responsible for mapping of predicate indicators to predicates,

and from there to ropes of clauses.

**Comment by jburse** [comment]
The change has been implemented, we also introduced the
prefix kb_ for the meta database and dynamic database retrieval.
Interestingly the dynamic database retrieval is ir_ agnostic,

since it uses exec_unify() and exec_build() so that in the interface
the result will by Prolog term and not intermediate representation.
We also introduced the sy_ prefix for system access, like the

wall clock. We event ported the JavaScript gc() and have now
not only wall clock but also garbage collection time. We could
compare JavaScript and Python, also in garbage collection Python 

is still a factor 30-40 slower, but nevertheless we managed to port it.
